modules/apps/frontend-js/frontend-js-web/test/liferay/util/Cookie.js

```diff
@@ -0,0 +1,213 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import Cookie from '../../../src/main/resources/META-INF/resources/liferay/util/cookie';
+
+describe('Liferay.Util.Cookie', () => {
+	const necessaryCookie = 'any-cookie';
+	const unnecessaryCookie = 'any-other-cookie';
+	const anyCookieValue = 'any-value';
+
+	const defaultCookieMap = {
+		[Cookie.TYPES.NECESSARY]: true,
+		[Cookie.TYPES.PERFORMANCE]: false,
+		[Cookie.TYPES.PERSONALIZATION]: false,
+		[Cookie.TYPES.FUNCTIONAL]: false,
+	};
+
+	beforeEach(() => {
+		Object.entries(defaultCookieMap).forEach(([name, value]) => {
+			document.cookie = `${name}=${value}`;
+		});
+	});
+
+	afterEach(() => {
+		jest.restoreAllMocks();
+		document.cookie = `${necessaryCookie}=; max-age=0`;
+		document.cookie = `${unnecessaryCookie}=; max-age=0`;
+	});
+
+	describe('Liferay.Util.Cookie.set', () => {
+		it('Always allows setting a necessary cookie', () => {
+			const cookieIsSet = Cookie.set(
+				necessaryCookie,
+				anyCookieValue,
+				Cookie.TYPES.NECESSARY
+			);
+
+			expect(cookieIsSet).toBe(true);
+
+			const setCookieValue = Cookie.get(necessaryCookie);
+			expect(setCookieValue).not.toBeUndefined();
+			expect(setCookieValue).toBe(anyCookieValue);
```

Comment:
We want en empty line between variable declaration and its usage 😉 

There are some similar situations below in the file in other tests.